pkg/webui/console/store/middleware/logics/gateway.js

```diff
@@ -14,26 +14,21 @@
 
 import { createLogic } from 'redux-logic'
 
-import sharedMessages from '../../../lib/shared-messages'
-import api from '../../api'
-import * as gateway from '../actions/gateway'
-import { gsConfigSelector } from '../../../lib/selectors/env'
-import { gatewaySelector } from '../selectors/gateway'
+import sharedMessages from '../../../../lib/shared-messages'
+import api from '../../../api'
+import * as gateway from '../../actions/gateway'
+import { gsConfigSelector } from '../../../../lib/selectors/env'
+import { gatewaySelector } from '../../selectors/gateway'
 import createEventsConnectLogics from './events'
+import createRequestLogic from './lib'
 
-const getGatewayLogic = createLogic({
+const getGatewayLogic = createRequestLogic({
   type: gateway.GET_GTW,
-  async process ({ action }, dispatch, done) {
-    const { id, meta = {}} = action
-    try {
-      const selectors = meta.selectors || ''
-      const gtw = await api.gateway.get(id, selectors)
-      dispatch(gateway.getGatewaySuccess(gtw))
-    } catch (error) {
-      dispatch(gateway.getGatewayFailure(error))
-    }
-
-    done()
+  async process ({ action }) {
+    const { id, meta = {}} = action.payload
```

Comment:
the structure of an action should be:
```
{
  type: ...,
  payload: ...,
  meta: ...
}
```
see my other comment on this

Reply:
I'm ok with putting it into meta, though it is not immediately obvious to me that request parameters should be put there. FSA docs are also not very clear on that. I agree having `meta` inside `payload` is obviously worse.